bw2data/parameters.py

```diff
@@ -1361,14 +1367,16 @@ def reformat(ds):
 
         with self.db.atomic():
             # Remove existing values
-            ProjectParameter.delete().where(ProjectParameter.name << tuple(new)).execute()
-            for idx in range(0, len(data), 100):
-                ProjectParameter.insert_many(data[idx : idx + 100]).execute()
+            # Call in loop to get event handling
+            for pp in ProjectParameter.select().where(ProjectParameter.name << tuple(new)):
+                pp.delete_instance()
+            for dataset in data:
+                ProjectParameter.create(**dataset)
             Group.get_or_create(name="project")[0].expire()
             ProjectParameter.recalculate()
 
     def new_database_parameters(self, data, database, overwrite=True):
-        """Efficiently and correctly enter multiple parameters. Deletes **all** existing database parameters for this database.
+        """Correctly enter multiple parameters. Deletes **all** existing database parameters for this database.
```

Comment:
Losing efficiency? :(

Reply:
Yes, we need to replace a single `delete` SQL statement with a loop as we only capture events on `.save()` and `.delete_instance()`. See [the Peewee docs](https://docs.peewee-orm.com/en/latest/peewee/playhouse.html#signals).